docs/sphinx/source/contributing.rst

```diff
@@ -73,17 +79,108 @@ environment.
+can help formalize this process.
+
+Of course, sometimes it is necessary to make a large pull request. We
+only ask that you take a few minutes to consider how to break it into
+smaller chunks before proceeding.
+
+pvlib-python contains :ref:`3 "layers" of code <modeling-paradigms>`:
+functions, PVSystem/Location, and ModelChain. We recommend that
+contributors focus their work on only one or two of those layers in a
+single pull request. New models are *not* required to be available to
+the higher-level API!
+
+
+When should I submit a pull request?
+------------------------------------
 
-The maintainers will follow same procedures, rather than making direct
-commits to the main repo. Exceptions may be made for extremely minor
-changes, such as fixing documentation typos.
+The short answer: anytime.
+
+The long answer: it depends. If in doubt, go ahead and submit. You do
+not need to make all of your changes before creating a pull request.
+Your pull requests will automatically be updated when you commit new
+changes and push them to GitHub.
+
+There are pros and cons to submitting incomplete pull-requests. On the
+plus side, it gives everybody an easy way to comment on the code and can
+make the process more efficient. On the minus side, it's easy for an
+incomplete pull request to grow into a multi-month saga that leaves
+everyone unhappy. If you submit an incomplete pull request, please be
+very clear about what you would like feedback on and what we should
+ignore. Alternatives to incomplete pull requests include posting a link
+to an experimental branch or to a `gist <https://gist.github.com>`_ in
+the corresponding issue.
+
+The best way to ensure that a pull request will be reviewed and merged in
+a timely manner is to:
+
+#. Start with an issue. The issue should be well-defined and actionable.
+#. Ask the maintainers to tag the issue with the appropriate milestone.
+#. Make a limited scope pull request. It can be a lot of work to check all of
+   the boxes in `pull request guidelines
+   <https://github.com/pvlib/pvlib-python/blob/master/.github/PULL_REQUEST_TEMPLATE.md>`_,
+   especially for pull requests with a lot of new primary code.
+   See :ref:`pull-request-scope`.
+#. Tag ``@pvlib/maintainer`` when the pull request is ready for review.
+
+Questions about related issues frequently come up in the process of
+addressing implementing code for a pull request. Please try to avoid
+expanding the scope of your pull request. We'd rather see small
+additions to the project's technical debt than see a pull request
+languish because its scope expanded beyond what the reviewer community
+is capable of processing.
+
+
+Code style
+~~~~~~~~~~
+
+pvlib python generally follows the `PEP 8 -- Style Guide for Python Code
+<https://www.python.org/dev/peps/pep-0008/>`_. Maximum line length for code
+is 79 characters.
+
+Documentation must be written in `numpydoc format <https://numpydoc.readthedocs.io/>`_.
```

Comment:
I didn't know about this specific format. Thanks!

My other big issue here has been knowing that my markup is going to look and link correctly in the eventual documentation.

Reply:
Added some text about building documentation on readthedocs. Not sure if it's sufficient since I can't actually walk through it because my account is already set up. I could add something about building it locally but it's somewhat cumbersome.